lib/peer_manager/modules/peer_data_storage.c

```diff
@@ -56,30 +61,20 @@ static void pds_evt_send(pm_evt_t *p_event)
 	}
 }
 
-/* Function to convert peer IDs to file IDs. */
-static uint16_t peer_id_to_file_id(pm_peer_id_t peer_id)
-{
-	return (uint16_t)(peer_id + PEER_ID_TO_FILE_ID);
-}
-
-/* Function to convert peer data id to type id. */
-static pm_peer_id_t file_id_to_peer_id(uint16_t file_id)
-{
-	return (pm_peer_id_t)(file_id + FILE_ID_TO_PEER_ID);
-}
-
-/* Function to convert peer data IDs to record keys. */
-static uint16_t peer_data_id_to_record_key(pm_peer_data_id_t peer_data_id)
+static uint32_t peer_id_peer_data_id_to_entry_id(pm_peer_id_t peer_id,
+						 pm_peer_data_id_t data_id)
 {
-	return (uint16_t)(peer_data_id + DATA_ID_TO_RECORD_KEY);
+	return (peer_id * PM_PEER_DATA_ID_LAST) + data_id;
```

Comment:
This mapping needs some future proofing IMO. `PM_PEER_DATA_ID_LAST` is liable to change in the future, which would break existing systems if it is updated with DFU. I suggest a bit-based division (shift the peer_id a certain number of bits, e.g. 16 since `peer_id_t` is a uint16_t).